mcuboot: kconfig: improve help

maass-hamburg opened the pull request:
improve kconfig help and make clear, that
MCUBOOT_GENERATE_UNSIGNED_IMAGE
will not lead to imgtool being used.

nordicjm reviewed (commented):
that's what `MCUBOOT_GENERATE_UNSIGNED_IMAGE` does

maass-hamburg commented:
> that's what `MCUBOOT_GENERATE_UNSIGNED_IMAGE` does

@de-nordic when `MCUBOOT_GENERATE_UNSIGNED_IMAGE` is set the script still runs and also runs imgtool, it is just executed without the key argument, when no key is provided.

maass-hamburg committed:
mcuboot: doc: improve kconfig help

improve kconfig help and make clear, that
MCUBOOT_GENERATE_UNSIGNED_IMAGE
will not lead to imgtool being used.

Signed-off-by: Fin Maaß <f.maass@vogl-electronic.com>

nordicjm commented on modules/Kconfig.mcuboot:
```diff
@@ -128,12 +130,21 @@ config MCUBOOT_EXTRA_IMGTOOL_ARGS
 	  string) you can use this option to pass extra options to
 	  imgtool. For example, you could set this to "--version 1.2".
 
+config MCUBOOT_SUPPRESS_WARNING_IMAGE_NOT_SIGNED
+	bool "Suppress warning about image not being signed"
+	depends on MCUBOOT_SIGNATURE_KEY_FILE = "" && !MCUBOOT_GENERATE_UNSIGNED_IMAGE
+	help
+	  If enabled, suppress the warning about the image not being signed/prepared
+	  when CONFIG_MCUBOOT_SIGNATURE_KEY_FILE is an empty string
+
```
we don't need this, it outputs a warning indicating the status which can be ignored if it's expected

maass-hamburg replied:
removed that, now only the change of the help text is lest
